### PR TITLE
Documentation updates

### PR DESCRIPTION
**Pull request by saghul**

### CODE REVIEW

File: docs/docs/projects.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+Reverse engineering framework.
+
+## [rquickjs](https://github.com/DelSkayn/rquickjs)
+
+High level Rust bindings.
+
+## [llrt](https://github.com/awslabs/llrt)
+
+Lightweight JS runtime for serverless applications.
```
Comment (TooTallNate) [inline docs/docs/projects.md:20]:
```suggestion
Lightweight JS runtime for serverless applications.

## [nx.js](https://github.com/TooTallNate/nx.js)

JavaScript runtime for Nintendo Switch homebrew applications.
```

Comment (saghul) [reply docs/docs/projects.md:20]: 🙌

File: docs/docs/cli.md [hunk 2]
```diff
@@ -75,3 +75,22 @@ options are:
...
+
+The resulting binary `hello` will be in the current directory.
+
+```bash
+$ ./hello
+Hello World
+```
+
+:::note
+We have plans to make this process easier, stay tunned!
```
Comment (bnoordhuis) [inline docs/docs/cli.md]:
```suggestion
We have plans to make this process easier, stay tuned!
```

File: docs/docs/diff.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
+# Differences with bellard/quickjs
+
+This project aims to be a drop-in replacement for those already using QuickJS.
+Minimal API changes might be necessary.
```
Comment (bnoordhuis) [inline docs/docs/diff.md:4]: We should probably flesh those out. The only one that comes to mind is class ids may need tweaking.

File: docs/docs/diff.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+## Performance
+
+While being an interpreter limits the performance in comparison with other engines which
+use a JIT, several significant performance improvements have been made:
+
+- Opcode fusion
+- Polymorphic inline caching
+- Memory allocation improvements
+- Improved variable resolver
```
Comment (bnoordhuis) [inline docs/docs/diff.md]:
```suggestion
- Improved parse speeds
```
I'm going to guess most people won't know what "variable resolver" means without more context.

File: docs/docs/diff.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+- Resizable ArrayBuffer
+- Float16Array
+- WeakRef
+- FinalizationRegistry
+- Iterator Helpers
+- Promise.try
+- Error.isError
+- Set operations
+
+Some non-standard APIs have also been added:
```
Comment (bnoordhuis) [inline docs/docs/diff.md]: Maybe:
```suggestion
Some non-standard but widely used APIs have also been added:
```

**Review by bnoordhuis — APPROVED** [review]

File: docs/docs/diff.md (hunk 3 above)
Comment (saghul) [reply docs/docs/diff.md:4]: - The malloc function changes
- SetOpaque can fail

Though those 2 are rare to see and maybe belong in advanced use cases...

Comment (saghul) [reply docs/docs/diff.md:4]: On my next pass I'll try and add some "migration" section.

**Pull request merged**